modules/deployment-scripts/create-agw-kv-certificate/main.bicep

```diff
@@ -0,0 +1,229 @@
+@description('The name of the Azure Key Vault')
+param akvName string
+
+@description('The name of the Azure Application Gateway')
+param agwName string = ''
+
+@description('The location to deploy the resources to')
+param location string = resourceGroup().location
+
+@description('How the deployment script should be forced to execute')
+param forceUpdateTag  string = utcNow()
+
+@description('Version of the Azure CLI to use')
+param azCliVersion string = '2.30.0'
+
+@description('Deployment Script timeout')
+param timeout string = 'PT30M'
+
+@description('The retention period for the deployment script')
+param retention string = 'P1D'
+
+@description('An array of Azure Key Vault RoleIds that are required for the DeploymentScript resource')
+param rbacRolesNeededOnKV array = [
```

Comment:
the edge case here is a slim one.
if custom roles are being used, having it as an array would allow multiple to be provided.

happy to scale this back down to a single role. 🚀 

multiple roles is more likely in the other modules i'm writing, but using arrays for RBAC is just my normal workflow now 🐱

Reply:
We can always expand later...

Reply:
YAGNI. 😉